Libraries/xcassets/Headers/xcassets/Asset/Asset.h

```diff
@@ -29,6 +29,7 @@ class Asset {
 private:
     FullyQualifiedName         _name;
     std::string                _path;
+    libutil::Filesystem const *_filesystem;
```

Comment:
What do you expect to change in the filesystem during the execution of the application?

I think it would be much better for Asset objects to have an API to return a byte vector or stream with the file contents. The only reason to know the path/filename is for the filename field in the CAR file which is useful for debugging.

I think it's awkward to have to pass around two arguments down to every function that works on assets, which is really each one that is not a stub, or not just a print contents action.